lib/intercepted_request_router.js

```diff
@@ -109,26 +117,38 @@ class InterceptedRequestRouter {
     })
   }
 
+  // from docs: When write function is called with empty string or buffer, it does nothing and waits for more input.
+  // However, actually implementation checks the state of finished and aborted before checking if the first arg is empty.
   handleWrite(buffer, encoding, callback) {
     debug('write', arguments)
     const { req } = this
 
-    if (!req.aborted) {
-      if (buffer) {
-        if (!Buffer.isBuffer(buffer)) {
-          buffer = Buffer.from(buffer, encoding)
-        }
-        this.requestBodyBuffers.push(buffer)
-      }
-      // can't use instanceof Function because some test runners
-      // run tests in vm.runInNewContext where Function is not same
-      // as that in the current context
-      // https://github.com/nock/nock/pull/1754#issuecomment-571531407
-      if (typeof callback === 'function') {
-        callback()
-      }
-    } else {
-      this.emitError(new Error('Request aborted'))
+    if (req.finished) {
+      const err = new Error('write after end')
+      err.code = 'ERR_STREAM_WRITE_AFTER_END'
+      this.emitError(err)
+      return true
```

Comment:
Should this be `return false`?

Reply:
🤷‍♂ Node returns `true` https://github.com/nodejs/node/blob/master/lib/_http_outgoing.js#L662-L665

Reply:
Cool, could you add a comment?